sdcm/cluster.py

```diff
@@ -2113,6 +2115,37 @@ def install_scylla(self, scylla_repo):
                 'sudo apt-get install -y -o Dpkg::Options::="--force-confdef" -o Dpkg::Options::="--force-confold" {}'
                 '--allow-unauthenticated {}'.format(force, self.scylla_pkg()))
 
+    def offline_install_scylla(self, unified_package, nonroot):
+        """
+        Offline install scylla by unified package.
+        """
+        # Download unified package
+        self.remoter.run(f'curl {unified_package} -o ./unified_package.tar.gz')
+
+        # Offline install does't provide openjdk-8, it has to be installed in advance
+        # https://github.com/scylladb/scylla-jmx/issues/127
+        if self.is_rhel_like():
+            self.remoter.run('sudo yum install -y java-1.8.0-openjdk')
+        else:
+            self.remoter.run('sudo apt-get install -y openjdk-8-jre-headless')
+            self.remoter.run('sudo update-java-alternatives --jre-headless -s java-1.8.0-openjdk-amd64')
+
+        if nonroot:
+            install_cmds = dedent("""
+                tar xvfz ./unified_package.tar.gz
+                ./install.sh --prefix=`pwd`/install_root
+                sudo rm -f /tmp/scylla.yaml
+            """)
+            # Known issue: https://github.com/scylladb/scylla/issues/7071
+            self.remoter.run('bash -cxe "%s"' % install_cmds)
```

Comment:
@roydahan  Currently nonroot will fail for known issue.
We don't what the right case will be.

After nonroot installation, if we can still manage scylla service by systemctl, the future change will be simple.
We have to leave this part until they fixed it.